scr/main/java/bittercode/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,98 @@
+package com.bittercode.service.impl;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+import javax.servlet.http.HttpSession;
+
+import com.bittercode.constant.ResponseCode;
+import com.bittercode.constant.db.UsersDBConstants;
+import com.bittercode.model.StoreException;
+import com.bittercode.model.User;
+import com.bittercode.model.UserRole;
+import com.bittercode.service.UserService;
+import com.bittercode.util.DBUtil;
+
+public class UserServiceImpl implements UserService {
+
+    private static final String registerUserQuery = "INSERT INTO " + UsersDBConstants.TABLE_USERS
+            + "  VALUES(?,?,?,?,?,?,?,?)";
+
+    private static final String loginUserQuery = "SELECT * FROM " + UsersDBConstants.TABLE_USERS + " WHERE "
+            + UsersDBConstants.COLUMN_USERNAME + "=? AND " + UsersDBConstants.COLUMN_PASSWORD + "=? AND "
+            + UsersDBConstants.COLUMN_USERTYPE + "=?";
+
+    @Override
+    public User login(UserRole role, String email, String password, HttpSession session) throws StoreException {
+        Connection con = DBUtil.getConnection();
+        PreparedStatement ps;
+        User user = null;
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Improve error handling in login method</b></div>
<div id="fix">

The `login` method might benefit from improved error handling. Consider wrapping the database operations in a try-with-resources block and handling `SQLException` more gracefully. Perhaps logging the error or throwing a custom exception could be more informative.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -27,26 +27,30 @@
      @Override
      public User login(UserRole role, String email, String password, HttpSession session) throws StoreException {
 -        Connection con = DBUtil.getConnection();
 -        PreparedStatement ps;
          User user = null;
 -        try {
 +        try (Connection con = DBUtil.getConnection();
 +             PreparedStatement ps = con.prepareStatement(loginUserQuery)) {
              String userType = UserRole.SELLER.equals(role) ? "1" : "2";
 -            ps = con.prepareStatement(loginUserQuery);
              ps.setString(1, email);
              ps.setString(2, password);
              ps.setString(3, userType);
 -            ResultSet rs = ps.executeQuery();
 -            if (rs.next()) {
 -                user = new User();
 -                user.setFirstName(rs.getString("firstName"));
 -                user.setLastName(rs.getString("lastName"));
 -                user.setPhone(rs.getLong("phone"));
 -                user.setEmailId(email);
 -                user.setPassword(password);
 -                session.setAttribute(role.toString(), user.getEmailId());
 +            try (ResultSet rs = ps.executeQuery()) {
 +                if (rs.next()) {
 +                    user = new User();
 +                    user.setFirstName(rs.getString("firstName"));
 +                    user.setLastName(rs.getString("lastName"));
 +                    user.setPhone(rs.getLong("phone"));
 +                    user.setEmailId(email);
 +                    user.setPassword(password);
 +                    session.setAttribute(role.toString(), user.getEmailId());
 +                }
              }
          } catch (SQLException e) {
 -            e.printStackTrace();
 +            logger.error("Error during login: ", e);
 +            throw new StoreException("Database error occurred during login", e);
          }
          return user;
      }
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/PrajaktaBendreBito/BookStore/pull/3#issuecomment-2434587212>#30653c</a></i></small>
</div>


---
🚩 Problem result:
- [ ] This issue was flagged incorrectly